### DISCUSSION TITLE
Does vLLM support flash attention?

### DISCUSSION THREAD

**Discussion by zhaoyang-star** [vllm-project/vllm#425] [Q&A]
Flash attention is an important optimizing method but I found no flash attention impls in vLLM code base. So does vLLM support flash attention?

**Comment by zhyncs** [comment]
> Flash attention is an important optimizing method but I found no flash attention impls in vLLM code base. So does vLLM support flash attention?

vLLM use [xformers](https://github.com/facebookresearch/xformers)'s [memory_efficient_attention_forward](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/attention.py#L84), so it makes indirect use of flash attention.
